streamer/pulp/streamer/adapters.py

```diff
@@ -212,22 +216,31 @@ def get_connection(self, url, proxies=None, verify=None, cert=None):
         :param url: The URL to connect to.
         :param proxies: (optional) A Requests-style dictionary of proxies used on this request.
         """
-        with self._pool_kw_lock:
-            if url.lower().startswith('https'):
-                self._update_poolmanager_ssl_kw(verify, cert)
+        scheme = urlparse(url).scheme.lower()
+
+        # The name of this lock is misleading and is protecting some
+        # very concerning behavior. The PoolManager and ProxyManager (IsA PoolManager)
+        # are thread safe. However, changing the PoolManager.connection_pool_kw content
+        # is not and could have dangerous side effects.  The critical section below
+        # does the following within the safety of the lock and must never be separated:
+        #   1. acquire the lock
+        #   2. update the manager properties (such as SSL certificates)
+        #   3. get a connection
+        #   4. release the lock
 
+        with self._pool_kw_lock:
             proxies = proxies or {}
-            proxy = proxies.get(urlparse(url.lower()).scheme)
+            proxy = proxies.get(scheme)
 
             if proxy:
                 proxy = prepend_scheme_if_needed(proxy, 'http')
-                proxy_manager = self.proxy_manager_for(proxy)
-                conn = proxy_manager.connection_from_url(url)
+                pool_manager = self.proxy_manager_for(proxy)
             else:
-                # Only scheme should be lower case
-                parsed = urlparse(url)
-                url = parsed.geturl()
```

Comment:
These two lines are interesting. I assume they're here just to normalize the URL, but it's unclear what value that provides. Maybe that helps with the pooling algorithm? In any case, since this came from upstream ``requests``, I think we should keep these two lines.